### DISCUSSION TITLE
4.4 Onion OS

### DISCUSSION THREAD

**Discussion by cac231** [OnionUI/Onion#1607] [Q&A] [answered]
What is the expected release date?

**Comment by schmurtzm** [comment, marked as answer]
No release date planned for now. We're running in slow motion at the moment, a well-deserved break I think.
I hope we'll find some time soon, because some of the 4.4 features are really great. Probably during the summer 🤞🏻

**Reply by cac231** [reply]
yes I understand, good luck and thank you

**Reply by gricelsepulveda** [reply]
We pray for a little patch with gpsp core (gba) last version please, the current included in 4.3 doesnt have netplay u.u

**Reply by Madeup1456** [reply]
Yes GBA wireless adapter support I would say is the biggest want right now.

**Comment by lbineau** [comment]
Dear Onion team,

Thank you for your amazing work on OnionOS, it is insanely good!
I have the same question 1 year later, do you have a rough idea when you want to realease the `4.4.0`?
I'm also very insterested about the GPSP and Netplay I so in the `4.4 beta` with a stable release :-)

Best regards.

**Reply by 0023-119** [reply]
holy moly it's been a year!

**Comment by Vamp898** [comment]
I would be willing to help, where can  i find the documentation for how to build the source and stuff?

**Reply by ThatStargirl** [reply]
Hey @Vamp898 fellow 'new-person-wanting-to-contribute' here, there is some documentation  on the website https://onionui.github.io/docs/dev I've not actually tried building it myself yet so can't guarantee its up to date, but its probably fine. A lot of the code is just shell script (ash, not bash) so you can get a long way just tinkering with that. 

For understanding how the system works, I found it helpful to:

 * Look at the installation zip files: runtime.sh is the file the myioo automatically runs that kicks off the install process. The install "MainUI" file is actually a shell script rather than a binary, despite having no extension.
 * Look at the file directories once installed, runtime.sh is different there and handles starting up Onion OS.
 * Each of the apps are defined in the app folder, you can make your own app by copying an existing one and messing with the config and associated launch.sh

The devs are still active retero game handhelds discord server https://onionui.github.io/docs/dev/contribute (though understandably less active than they were historically I think) and have been kind enough to answer some of my questions and even provide a diagram for what the startup looks like. 

Feel free to reach out to me too there and I'll be happy to share anything I've figured out. With the original devs being away from the project for now, us 'contributor-hopefuls' should probably organise ourselves a little!
